### PR TITLE
[EventHubs] Partition Receiver: GetPartitionPropertiesAsync (Track Two)

### PR DESCRIPTION
**Pull request by kinelski**
Included partial implementation and tests for the class `PartitionReceiver`. Names are not final and class is `internal` for the time being.

Changes:
- `GetPartitionPropertiesAsync` method.

This implementation is a simple adaptation of the `PartitionReceiver` that was present during previews and follows the [skeleton](https://gist.github.com/jsquire/4ea3368192cfee98dcbb4adfb29a9822) that will be used as a base for the final public API this class will have.

Related: 

- #9323 
- [Partition Receiver Design Proposal](https://gist.github.com/jsquire/4ea3368192cfee98dcbb4adfb29a9822)

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs [hunk 1]
```diff
@@ -242,6 +242,21 @@ protected PartitionReceiver()
...
+        ///
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>The set of information for the associated partition under the Event Hub this client is associated with.</returns>
+        ///
+        public virtual Task<PartitionProperties> GetPartitionPropertiesAsync(CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(PartitionReceiver));
+            return Connection.GetPartitionPropertiesAsync(PartitionId, RetryPolicy, cancellationToken);
+        }
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs]: Once again, I'm copying the behavior from other clients. @jsquire I remember you mentioning it was preferred to make the method `async` and calling `await` on the connection call (at least in that specific `EventProcessor` method). Is this something we want to do here as well? I can't really remember the reasoning behind it.

Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs]: We would need to update other clients if that's the case.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs [hunk 2]
```diff
@@ -1953,11 +1953,10 @@ await using (var consumer = new EventHubConsumerClient(EventHubConsumerClient.De
         [TestCase(EventHubsTransportType.AmqpWebSockets)]
         public async Task ConsumerCanRetrievePartitionProperties(EventHubsTransportType transportType)
         {
-            var partitionCount = 4;
+            var partitionCount = 1;
```
Comment (ellismg) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs]: For my own understanding - Why the changes to this file?  My guess is just general cleanup (I see how this test is very similar to the new tests being adder for the PartitionReceiver) and here we don't actually need more than a single partition - but I'm wondering if there was something more subtle going on?  If it's just cleaning up code - Nice Cleanup!

**Review by ellismg — APPROVED** [review]
LGTM

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs [hunk 3]
```diff
@@ -242,6 +242,21 @@ protected PartitionReceiver()
...
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>The set of information for the associated partition under the Event Hub this client is associated with.</returns>
+        ///
+        public virtual Task<PartitionProperties> GetPartitionPropertiesAsync(CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(PartitionReceiver));
+            return Connection.GetPartitionPropertiesAsync(PartitionId, RetryPolicy, cancellationToken);
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs]: It has been recommended that we `await` these rather than returning the raw task.   _(see: [David Fowler's guidance](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task))_.

**Review by jsquire — APPROVED** [review]
LGTM.    Just one small comment.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs (hunk 2 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs]: It's just cleanup. Like you said, we were using only one partition.

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs (hunk 3 above)
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs]: Actually that's what I was asking about. I'll update other clients as well.
